torch/onnx/_internal/fx/type_utils.py

```diff
@@ -1,11 +1,35 @@
 """Utilities for converting and operating on ONNX, JIT and torch types."""
 from __future__ import annotations
 
-from typing import Any, Dict, List, Optional, Set, Tuple, Union
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Protocol,
+    runtime_checkable,
+    Set,
+    Tuple,
+    Union,
+)
 
 import torch
 
 
+# Enable both TorchScriptTensor and torch.Tensor to be tested
+# for dtype in OpSchemaWrapper.
+@runtime_checkable
+class TensorLike(Protocol):
+    @property
+    def dtype(self) -> Optional[torch.dtype]:
+        ...
+
+
+def is_torch_complex_dtype(tensor: TensorLike) -> bool:
+    # NOTE: torch.Tensor can use torch.is_complex() to check if it's a complex tensor
```

Comment:
Is this a recommendation/alternative or is it trying to explain why `tensor.dtype in _COMPLEX_TO_FLOAT` is needed? I would clarify

Reply:
Added comments